pageserver/src/tenant/mgr.rs

```diff
@@ -522,97 +578,102 @@ pub async fn init_tenant_mgr(
             }
         };
 
-        let generation = if let Some(generations) = &tenant_generations {
+        // FIXME: if we were attached, and get demoted to secondary on re-attach, we
+        // don't have a place to get a config.
+        // (https://github.com/neondatabase/neon/issues/5377)
+        const DEFAULT_SECONDARY_CONF: SecondaryLocationConfig =
+            SecondaryLocationConfig { warm: true };
```

Comment:
Wouldn't it be more straightfoward if we returned the full location config on re-attach? It would solve this case at least.

Reply:
Yeah: we might end up doing that in future -- this PR is an incremental improvement.  The reason for not jumping straight to that solution is general paranoia about how big the list of configs for 10k+ tenants gets when we send it down as one JSON response.